server/server.go

```diff
@@ -22,17 +22,18 @@ func Start(addr string, s Store) error {
 }
 
 type IndexHandler struct {
-	store Store
+	store data.Store
 }
 
-func NewIndexHandler(s Store) *IndexHandler {
+func NewIndexHandler(s *data.Store) *IndexHandler {
 	return &IndexHandler{
-		store: s,
+		store: *s,
```

Comment:
Dereferencing here makes a copy of the `Store`. Likely this is not what is intended.

Reply:
Not at all, will need to fix this.